torchmetrics/functional/detection/panoptic_quality.py

```diff
@@ -0,0 +1,235 @@
+        iou = intersection / union
+
+        if iou > 0.5:
+            pred_segment_matched.add(pred_color)
+            target_segment_matched.add(target_color)
+            iou_sum[continuous_id] += iou
+            true_positives[continuous_id] += 1
+
+    # count false negative: ground truth but not matched
+    # areas that are mostly void in the prediction are ignored
+    false_negative_colors = set(target_areas.keys()).difference(target_segment_matched)
+    false_negative_colors.discard(void_color)
+    for target_color in false_negative_colors:
+        void_target_area = intersection_areas.get((void_color, target_color), 0)
+        if void_target_area / target_areas[target_color] > 0.5:
+            continue
+        continuous_id = cat_id_to_continuous_id[target_color[0]]
+        false_negatives[continuous_id] += 1
+
+    # count false positive: predicted but not matched
+    # areas that are mostly void in the target are ignored
+    false_positive_colors = set(pred_areas.keys()).difference(pred_segment_matched)
+    false_positive_colors.discard(void_color)
+    for pred_color in false_positive_colors:
+        pred_void_area = intersection_areas.get((pred_color, void_color), 0)
+        if pred_void_area / pred_areas[pred_color] > 0.5:
+            continue
+        continuous_id = cat_id_to_continuous_id[pred_color[0]]
+        false_positives[continuous_id] += 1
+
+    return iou_sum, true_positives, false_positives, false_negatives
+
+
+def _panoptic_quality_compute(
+    things: Set[int],
+    stuff: Set[int],
+    iou_sum: torch.Tensor,
+    true_positives: torch.Tensor,
+    false_positives: torch.Tensor,
+    false_negatives: torch.Tensor,
+) -> Dict:
+    # TODO: exclude from mean categories that are never seen ?
+    # TODO: per class metrics
+
+    # per category calculation
+    denominator = (true_positives + 0.5 * false_positives + 0.5 * false_negatives).double()
+    panoptic_quality = torch.where(denominator > 0.0, iou_sum / denominator, 0.0)
+    segmentation_quality = torch.where(true_positives > 0.0, iou_sum / true_positives, 0.0)
+    recognition_quality = torch.where(denominator > 0.0, true_positives / denominator, 0.0)
+
+    metrics = dict(
+        all=dict(
+            pq=torch.mean(panoptic_quality),
+            rq=torch.mean(recognition_quality),
+            sq=torch.mean(segmentation_quality),
+            n=len(things) + len(stuff),
+        ),
+        things=dict(
+            pq=torch.mean(panoptic_quality[: len(things)]),
+            rq=torch.mean(recognition_quality[: len(things)]),
+            sq=torch.mean(segmentation_quality[: len(things)]),
+            n=len(things),
+        ),
+        stuff=dict(
+            pq=torch.mean(panoptic_quality[len(things) :]),
+            rq=torch.mean(recognition_quality[len(things) :]),
+            sq=torch.mean(segmentation_quality[len(things) :]),
+            n=len(stuff),
+        ),
+    )
```

Comment:
I do not understand why a dict is constructed with the `all`, `things`, `stuff` keys initialized, but what is actually returned to the user is just: `results["all"]["pq"]`.
Seems it would be more efficient to just return `torch.mean(panoptic_quality)` directly